### PR TITLE
refac: redefine builders as create methods in AppBio QuantStudio

### PR DESCRIPTION
**Pull request by alejandro-salgado**
- Refactor builders as create methods in AppBio QuantStudio

* first part at https://github.com/Benchling-Open-Source/allotropy/pull/128
* second part of https://github.com/Benchling-Open-Source/allotropy/pull/38

### CODE REVIEW

File: src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_structure.py [hunk 1]
```diff
@@ -165,13 +469,200 @@ class Result:
...
+
 
 @dataclass
 class MeltCurveRawData:
     reading: list[float]
     fluorescence: list[Optional[float]]
     derivative: list[Optional[float]]
 
+    @staticmethod
+    def build(data: pd.DataFrame, well: Well) -> MeltCurveRawData:
```
Comment (nathan-stender) [inline src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_structure.py:646]:
```suggestion
    def create(data: pd.DataFrame, well: Well) -> MeltCurveRawData:
```

File: src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_data_creator.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    MulticomponentData,
+    RawData,
+    Result,
+    WellList,
+)
+from allotropy.parsers.lines_reader import LinesReader
+from allotropy.parsers.utils.values import try_str_from_series_or_none
+
+
+def create_data(reader: LinesReader) -> Data:
```
Comment (nathan-stender) [inline src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_data_creator.py:19]: Please go ahead and put this as Data.create method in the structure file.  

This function is not over-large, and I think after this refactor there will be some other patterns we can identify to simplify further :D 

For future reference, if we were going to break up a file, I would lean towards just breaking up large Structure classes into their own files. But in this case, I think it's OK to be in a single file.

**Review by nathan-stender — CHANGES_REQUESTED** [review]

File: src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_data_creator.py (hunk 2 above)
Comment (alejandro-salgado) [reply src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_data_creator.py:19]: I tried to move this function into a Data.create function. But I was not able to fix a circular reference issue that results.

The problem is that this method creates the calculated data documents and those use the other structure classes
So, If I move this code to the structure a circular reference is created.

Comment (nathan-stender) [reply src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_data_creator.py:19]: Ah ok, I thought it was just due to size. I still think we can make it work, but OK with merging this and taking a crack at it later. 👍

**Review by nathan-stender — APPROVED** [review]

**Comment by nathan-stender** [comment]
Thanks again for all the hard working improving code consistency!

**Pull request merged**